Add /vanish and /invsee

Commodore64x opened the pull request:
PR for issues #2022 and #2018
Enjoy!

Commit:
Added invsee command

Commit:
Added Vanish Command

Commit:
Added vanish and invsee variables

Commit:
Added in methods and events to handle vanish and invsee

mayokek commented:
As per the contributing guidelines, you may make a PR that solves an issue, not multiple.
```
Due to a lot of pull requests, and the effort it takes to review each PR, some requirements for pull requests have been imposed. PRs opened that do not meet the pull request requirements may be closed.

PRs must only address issues which are marked as "accepted".
PRs must only address one issue per PR, not several.
PRs must not solely be grammar fixes, config-edits, small changes (less than 10 lines of code) or message rephrasing edits. Small edits such as these take very little effort to make and a lot more effort to review and integrate into the codebase.
PRs must be compiled and tested before they are submitted. If your PR does not function or has clearly not been tested, it may be closed without notice.
```

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
         return false;
     }
+
+    public Boolean isVanished(String username)
```
Use lowercase b

Commodore64x replied:
No, this is correct.

Commodore64x replied:
Quoting stackoverflow: "boolean is a java primitive type whereas Boolean is an object/reference type that wraps a boolean" To return null it has to be an Object, because boolean can only return true or false.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+
+    public void setVanished(String username, boolean vanished)
+    {
+
```
White line should be removed from here.

Wild1145 replied:
Agreed

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+        {
+            user.setVanished(vanished);
+        }
+
```
White line should be removed from here.

Wild1145 replied:
Agreed

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:
```diff
@@ -0,0 +1,70 @@
+
+        return true;
+    }
+}
```
Missing an extra line

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+            final Player player = refreshPlayer;
+            new BukkitRunnable()
+            {
+
```
White line should be removed from here.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+            final Player player = refreshPlayer;
+            new BukkitRunnable()
+            {
+
```
White line should be removed from here.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+                {
+                    player.updateInventory();
+                }
+
```
White line should be removed from here.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+        final FPlayer fPlayer = plugin.pl.getPlayer(player);
+        if (fPlayer.isVanish())
+        {
+
```
White line should be removed from here.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+
```
White line should be removed from here.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:
```diff
@@ -0,0 +1,54 @@
+        {
+            msg("You can't spy on admins!");
+            return true;
+
```
White line should be removed from here.

Wild1145 replied:
Agreed

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:
```diff
@@ -0,0 +1,54 @@
+        playerSender.openInventory(playerInv);
+        return true;
+    }
+
```
White line should be removed from here.

mayokek commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:
```diff
@@ -0,0 +1,54 @@
+        return true;
+    }
+
+}
```
Missing an extra line after this

Wild1145 replied:
What do you mean "Missing an extra line"?

mayokek replied:
Unless I am mistaken, At the end of the code, there always a new line.

Wild1145 replied:
There shouldnt be generally.

mayokek replied:
You're correct.

mayokek commented:
I should've use review changes instead of leaving single-comment

ghost commented:
mayokek, no offence but that review is completely useless. We are more relaxed about this stuff now and an extra line of whitespace does not hurt, neither does putting two very similar issues into one PR.

mayokek commented:
Formatting matters when you are coding. And also, how does those 2 issues similar?

ghost commented:
There is nothing wrong with the formatting.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
This should probably be moved into a listener class and not in the bridge its self.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
Again, this should be in a listener file really.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
Again, should be in a listener file.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
Also in a listener file

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:
```diff
@@ -0,0 +1,70 @@
+import org.bukkit.potion.PotionEffectType;
+
+@CommandPermissions(level = Rank.SUPER_ADMIN, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Vanish yourself.", usage = "/<command>", aliases = "v")
```
Not sure we want aliases... Im sure we had the discussion before and wanted to avoid them in order to keep things simple.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:
```diff
@@ -0,0 +1,54 @@
+import org.bukkit.inventory.Inventory;
+
+@CommandPermissions(level = Rank.OP, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Look into another player's inventory, optionally take items out.", usage = "/<command> <player>", aliases = "inv,insee")
```
Also not sure we wanted aliases. Especially not multiple ones.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:
```diff
@@ -0,0 +1,70 @@
+                player.hidePlayer(playerSender);
+            }
+            plugin.esb.setVanished(playerSender.getName(), true);
+            playerSender.addPotionEffect(new PotionEffect(PotionEffectType.INVISIBILITY, 1000000, 1000000, true, false));
```
Is there not a better way of doing this? Like GameMode 3?

Commodore64x replied:
The whole idea of this command is to completely vanish but retain their current gamemode, if users wish to do this then they can go into that gamemode themselves

Wild1145 replied:
I guess my concern is this is not completely vanished though. Players can still be seen with this both through hacked clients and having particle effects enabled?

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:
```diff
@@ -0,0 +1,70 @@
+package me.totalfreedom.totalfreedommod.command;
```
You should probably also update the /list command, as invis players should still show in that because TFM will not check if they are invis currently.

Commodore64x replied:
Done

Wild1145 commented:
More generally in regards to formatting, while its not the end of the world, if you configure your IDE once properly, you can just format the files as you're working on them, and before you commit them. Formatting will need to be corrected, but I think given these are smaller changes we can be flexible in not needing two different PR's on this occasion.

Commit:
Update /list to hide vanished users

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:
```diff
@@ -33,18 +31,6 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
             return false;
         }
-
-        if (FUtil.isFromHostConsole(sender.getName()))
```
Why did you remove this?

Commodore64x replied:
oh rip, wrong code

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:
```diff
@@ -68,31 +54,54 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
 
+        if (listFilter == ListFilter.VANISHED_ADMINS && !isAdmin(playerSender))
+        {
+            msg("You can't view vanished admins.");
```
I probably wouldnt even list this, I'd just make it return an unkown command with the syntax.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:
```diff
@@ -2,27 +2,25 @@
 
 @CommandPermissions(level = Rank.IMPOSTOR, source = SourceType.BOTH)
-@CommandParameters(description = "Lists the real names of all online players.", usage = "/<command> [-a | -i | -f]", aliases = "who")
+@CommandParameters(description = "Lists the real names of all online players.", usage = "/<command> [-a | -i | -v]", aliases = "who")
```
Not sure I'd add -v to the syntax, if only admins know invis admins are a thing its no bad thing.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:
```diff
@@ -2,27 +2,25 @@
-
         PLAYERS,
         ADMINS,
-        FAMOUS_PLAYERS,
```
Why remove famous players? While I didnt know it was a thing, im not sure why we're removing the functionality?

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:
```diff
@@ -68,31 +54,54 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
                 continue;
             }
 
-            if (listFilter == ListFilter.IMPOSTORS && !plugin.al.isAdminImpostor(player))
```
Why remove imposters from the list?

Commit:
Take 2

Commit:
Removed aliases

Commit:
Remove aliases

Commit:
Removed Invsee and Vanish Listeners

Commit:
New file just for Vanish listeners

Commit:
Invsee listener file

Commit:
Added services to TFM

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Override
+    protected void onStart()
```
Im not sure why you need this, it does nothing?

lemonsked replied:
Aero requires it, I think, for services.

mayokek replied:
Lemon is correct. `FreedomService` requires the method to be added

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Override
+    protected void onStop()
```
Im not sure why you need this, it does nothing?

lemonsked replied:
Aero requires it, I think, for services.

mayokek replied:
Lemon is correct. `FreedomService` requires the method to be added

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:
```diff
@@ -136,4 +137,28 @@ public boolean isEssentialsEnabled()
+        }
+
+    }
+
```
Also if you could remove this line as well

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:
```diff
@@ -0,0 +1,54 @@
+
+        }
+
+        playerSender.closeInventory();
```
Why are you closing the original players inventory? Does it then not give it away that you're looking in it?

mayokek replied:
@Wild1145, the `playerSender` method is still sender that have been converted to player.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:
```diff
@@ -101,7 +134,8 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
         onlineUsers.append("Connected ");
-        onlineUsers.append(playerType + ": ");
+        onlineUsers.append(playerType);
+        onlineUsers.append(": ");
```
Why have you split this into two lines?

Commodore64x replied:
Because you don't concatenate in .append, you append it.

mayokek replied:
This can be merge into one line.

Wild1145 commented on src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:
```diff
@@ -0,0 +1,70 @@
+            {
+                loginMsg = FUtil.colorize(admin.getLoginMessage());
+            }
+            FUtil.bcastMsg(ChatColor.AQUA + playerSender.getName() + " is " + loginMsg);
```
Im not sure on this. Is this not different depending on the rank / title the admin has?

lemonsked replied:
Nope, it isn't.

Commit:
Removed Whitespacing

Commit:
Removed whitespacing

Commit:
Now sets you to spectator

ghost commented:
All requested changes have been implemented

mayokek commented:
@Commodore64x, I don't believe spectator is a good idea because if we need to interact with items, we must go out of vanish.

ghost commented:
Wild asked for it to be changed, regardless you can go into creative anyway

Wild1145 commented:
@Commodore64x @mayokek Spectator may not be the best way to implement it, the honest answer is I dont know what the absolute best way is. Spectator sounds like it'll be the best because its totally invisible. You should not need to interact with blocks and similar when in /invis anyway??

ghost commented:
/invis doesn't make you invisible

Wild1145 commented:
@Commodore64x Have I misunderstood the purpose of /invis?

Wild1145 commented:
@Commodore64x Sorry, My mistake, I did not mean /invis but /vanish - My mistake, its been a long day. I was in fact referring to /vanish when I was asking about people being made invisible.

Commit:
Remove change of gamemode

Changing gamemode or adding potions effects is not needed! You are not visible to any other players anyway.

ghost commented:
@Wild1145 Can you remove the Needs Changes tag? There are no outstanding changes that haven't been fixed.

OxLemonxO commented on src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:
```diff
@@ -0,0 +1,106 @@
+    @Override
+    protected void onStart()
+    {
+
```
Remove whitespace.

lemonsked replied:
Remove whitespace.

lemonsked commented on src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:
```diff
@@ -0,0 +1,106 @@
+    @Override
+    protected void onStop()
+    {
+
```
Remvoe whitespace.

lemonsked commented on src/main/java/me/totalfreedom/totalfreedommod/Vanish.java:
```diff
@@ -0,0 +1,66 @@
+        final FPlayer fPlayer = plugin.pl.getPlayer(player);
+        if (fPlayer.isVanish())
+        {
+
```
Remove extra whitespace.

Commit:
Frontdoor IP update (#2064)

Wild1145 committed:
Increased text limit. Resolves #1941 (#2039)



* Updated the comment as I forgot that

* Corrected the max limited based on updated issue from 250 to 256

Commit:
Update /smite (#2009)

This update makes smite messages stand out more in chat and to the smitten player.

Wild1145 committed:
Corrected player message for /disguisetoggle. Resolves #1872 (#2042)

Wild1145 committed:
Added a catch for if a player is caged and kicked. Resolves #1789 (#2041)

Commit:
Vanished users no longer display in radar

Wild1145 reviewed (changes requested):
This is currently not building for me. Can you take a look at this for me please?

Commit:
Fixed import error

Wild1145 commented:
@Commodore64x This is still failing the TravisCI Integration tests.

lemonsked commented:
@Wild1145  For some reason, the TravisCI integration tests think that the language is ruby, so it fails.
https://travis-ci.org/TotalFreedom/TotalFreedomMod/jobs/257602861/config

Wild1145 commented:
@OxLemonxO That'd do it.

JeromSar commented:
@Wild1145 @OxLemonxO The TotalFreedomMod `devel` branch now contains the required _.travis.yml_ file for correct Travis CI integration. @Commodore64x Merge this branch with the upstream `devel` branch to fix the issue.

Commit:
Wild's changes (#2)

* Moved all dependencies over to proper resoultion. Currently experiencing an issue with TF-WorldEdit not pulling custom additions

* Final fix to the dependency resolution. Reverted minor changes to WorldEdit handling

* Added Travis YAML File for CI#

* Fixed issues with POM file - Everything is resolving properly and CI is working

* Update pom.xml

Updated Libs Disguise version

* Update pom.xml

* Updated Libs Disguise Version Dependency

It will now always pull the latest 1.11 branch commit, this should remain build-able and stable as it is a TF Project.

* Update README.md

* Updated to 1.12 R01 for dependencies

* Updated dependencies removing Libs Disguise due to unstable upstream repos

* Resolved dependency issues - Building correctly

* Removed dependency to Craftbukkit and Spigot server code.

This breaks whitelist and onlinemode toggle.

* Add .travis.yml file for Travis CI integration

* Update Travis CI status image link.

* Logviewer registration overhaul.

* Set up Travis CI automatic build uploads to SFTP server.

ghost commented:
@JeromSar @Wild1145 Should work with travis now.

Wild1145 commented:
@Commodore64x Im not sure what you've done, but you've done something really weird with the Git History there...
